lightning/src/util/events.rs

```diff
@@ -145,6 +146,8 @@ pub enum Event {
 		/// failed. This will be set to false if (1) this is an MPP payment and (2) other parts of the
 		/// larger MPP payment were still in flight when this event was generated.
 		all_paths_failed: bool,
+		/// The payment path that failed.
+		path: Vec<RouteHop>,
```

Comment:
So this will be empty for backwards compatibility rather than `None`? I think that's probably ok but just want to make sure.

Reply:
True. Is that an issue for the scorer? It could be an `option`

Reply:
Yes, `vec_type` is an option, basically. I agree I think its fine, but should be called out in release notes.

Reply:
> True. Is that an issue for the scorer? It could be an `option`

Nah, would be better if neither events used `Option` for symmetry.

> Yes, `vec_type` is an option, basically. I agree I think its fine, but should be called out in release notes.

Do we tag PRs that need to be reflected in the release? Or should we update as we go?

Reply:
We currently don't have a process for release notes, I generally write the compat section by reading the full-release diff and looking at serialization sections to see the implications of stuff when we cut the release. In the future, we should move towards including release note updates in the PR itself, maybe in a subfolder that just has a long list of release note files. Either way, doesn't have to happen here.